x509-cert/src/name.rs

```diff
@@ -2,17 +2,181 @@
+        self.by_oid::<PrintableStringRef<'_>>(rfc4519::COMMON_NAME)
+    }
+
+    /// Returns the Country (C) found in the name.
+    ///
+    /// This will return a `Ok(None)` if no Country is found.
+    ///
+    /// # Errors
+    ///
+    /// This will return a [`der::Error`] if the content is not serialized as a printableString.
+    pub fn country(&self) -> der::Result<Option<PrintableStringRef<'_>>> {
+        self.by_oid::<PrintableStringRef<'_>>(rfc4519::COUNTRY)
+    }
+
+    /// Returns the State or Province (ST) found in the name.
+    ///
+    /// This will return a `Ok(None)` if no State or Province is found.
+    ///
+    /// # Errors
+    ///
+    /// This will return a [`der::Error`] if the content is not serialized as a printableString.
+    pub fn state_or_province(&self) -> der::Result<Option<PrintableStringRef<'_>>> {
+        self.by_oid::<PrintableStringRef<'_>>(rfc4519::ST)
+    }
+
+    /// Returns the Locality (L) found in the name.
+    ///
+    /// This will return a `Ok(None)` if no Locality is found.
+    ///
+    /// # Errors
+    ///
+    /// This will return a [`der::Error`] if the content is not serialized as a printableString.
+    pub fn locality(&self) -> der::Result<Option<PrintableStringRef<'_>>> {
+        self.by_oid::<PrintableStringRef<'_>>(rfc4519::LOCALITY)
+    }
+
+    /// Returns the Organization (O) found in the name.
+    ///
+    /// This will return a `Ok(None)` if no Organization is found.
+    ///
+    /// # Errors
+    ///
+    /// This will return a [`der::Error`] if the content is not serialized as a printableString.
+    pub fn organization(&self) -> der::Result<Option<PrintableStringRef<'_>>> {
+        self.by_oid::<PrintableStringRef<'_>>(rfc4519::ORGANIZATION)
+    }
+
+    /// Returns the Organization Unit (OU) found in the name.
+    ///
+    /// This will return a `Ok(None)` if no Organization Unit is found.
+    ///
+    /// # Errors
+    ///
+    /// This will return a [`der::Error`] if the content is not serialized as a printableString.
+    pub fn organization_unit(&self) -> der::Result<Option<PrintableStringRef<'_>>> {
+        self.by_oid::<PrintableStringRef<'_>>(rfc4519::ORGANIZATIONAL_UNIT)
+    }
+
+    /// Returns the Email Address (emailAddress) found in the name.
+    ///
+    /// This will return a `Ok(None)` if no email address is found.
+    ///
+    /// # Errors
+    ///
+    /// This will return a [`der::Error`] if the content is not serialized as a printableString.
+    pub fn email_address(&self) -> der::Result<Option<PrintableStringRef<'_>>> {
+        self.by_oid::<PrintableStringRef<'_>>(rfc3280::EMAIL_ADDRESS)
+    }
+}
+
+/// Parse a [`Name`] string.
+///
+/// Follows the rules in [RFC 4514].
+///
+/// [RFC 4514]: https://datatracker.ietf.org/doc/html/rfc4514
+impl FromStr for Name {
+    type Err = der::Error;
+
+    fn from_str(s: &str) -> der::Result<Self> {
+        Ok(Self(RdnSequence::from_str(s)?))
```

Comment:
Are there more rules that should be enforced here?

Reply:
I don't see any. There are more in CABF, but this is enforced in the cert builder.